### PR TITLE
de-bitrot the BSDs

### PR DESCRIPTION
**Pull request by mikdusan**
Advance the state of various BSDs ports on x86_64 to be able to build stage3 and pass `zig test lib/std/std.zig`:

- netbsd 9.3
- openbsd 7.3
- dragonfly 6.4
- freebsd 13.2 (wasn't broken but tested anyways)

### CODE REVIEW

File: lib/std/c.zig [hunk 1]
```diff
@@ -408,11 +408,17 @@ pub extern "c" fn setlogmask(maskpri: c_int) c_int;
...
+        // musl does not implement getcontext
+        pub const getcontext = std.os.linux.getcontext;
+    } else struct {
+        pub extern "c" fn getcontext(ucp: *std.os.ucontext_t) c_int;
+    },
+    .openbsd => struct {},
+    else => struct {
+        pub extern "c" fn getcontext(ucp: *std.os.ucontext_t) c_int;
+    },
 };
```
Comment (andrewrk) [inline lib/std/c.zig:422]:
```suggestion
pub extern "c" fn getcontext(ucp: *std.os.ucontext_t) c_int;
```

 * don't fear linker errors
 * std.c should not re-export something from std.os.linux
 * avoid `usingnamespace`

**Review by andrewrk — CHANGES_REQUESTED** [review]
Thanks! Just one suggestion before this is ready.

**Comment by mikdusan** [comment]
> Thanks! Just one suggestion before this is ready.

Done. There were a few consequences in `std.debug` but handled.

If you sign off, before merge, I need to update the commit log and drop the OpenBSD commit (no longer needed).

File: lib/std/debug.zig [hunk 2]
```diff
@@ -167,29 +158,44 @@ pub fn relocateContext(context: *ThreadContext) void {
...
+            // choose std.c implementation
+            break :nofpt .{ .enabled = true, .getcontext = std.c.getcontext };
+        }
+    } else if ((builtin.os.tag == .linux) and @hasDecl(os.linux.arch_bits, "getcontext")) {
+        // choose syscall implementation
+        break :nofpt .{ .enabled = true, .getcontext = os.linux.arch_bits.getcontext };
+    } else {
+        break :nofpt .{ .enabled = false };
+    }
+};
```
Comment (andrewrk) [inline lib/std/debug.zig:186]: This function should not exist.

Note that `@import("builtin").omit_frame_pointer` exists.

**Review by andrewrk — CHANGES_REQUESTED** [review]

**Review by andrewrk — APPROVED** [review]
Looks great, I'm happy with where this landed, thanks!

**Pull request merged**